src/kmer.jl

```diff
@@ -0,0 +1,31 @@
+# kmer.jl
+# =======
+#
+# Basic kmer types.
+#
+# This file is a part of the Kmers.jl, a package in the BioJulia ecosystem.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+abstract AbstractKmer
+
+for inttype in subtypes(Unsigned)
+    nbits = sizeof(inttype) * 8
+    typename = "Kmer$nbits"
+    @eval begin
+        bitstype $nbits $(parse(typename)){T<:NucleicAcid, K} <: AbstractKmer
```

Comment:
Since we're going to drop the support of Julia 0.5 in this package (right?), we should use the `primitive type ... end` syntax for bit-type definitions.